fix(migration): detail the Oracle version change

jjargot opened the pull request:
Elaborate on the Oracle version change needed before using the migration script to migrate to 7.9.0.
Indeed, many customer faced issues with final target versions.

Versions: 7.9, 7.10

jjargot committed:
fix(migration): detail the Oracle version change

Elaborate on the Oracle version change needed before using the migration script to migrate to 7.9.0.
Indeed, many customer faced issues with final target versions.

Versions: 7.9, 7.10

jjargot committed:
fix(migration): detail the Oracle version change

fix typo 01

jjargot committed:
fix(migration): detail the Oracle version change


fix 02

jjargot committed:
fix(migration): detail the Oracle version change


fix 03

jjargot committed:
fix(migration): detail the Oracle version change


fix 04

ugaston commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -298,13 +298,35 @@ please follow this procedure:
-* then upgrade your Oracle database server to the version 12c (it must be 12.2.x.y)
-* in a second step, run the migration tool again to migrate the platform to 7.9.0 or newer
-* once done, you can restart your updated Bonita platform
+Bonita 7.9+ supports Oracle 12c (12.2.x.y) version: this is a requirement change.
```
```suggestion
Bonita 7.9+ supports Oracle 12c (12.2.0.x.y) version: this is a requirement change.
```

ugaston commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -298,13 +298,35 @@ please follow this procedure:
+### Upgrade Oracle database server
+
+* shut down the Bonita platform
+* upgrade the Oracle database server to the version 12c (it must be 12.2.x.y)
```
```suggestion
* upgrade the Oracle database server to the version 12c (it must be 12.2.0.x.y)
```

ugaston commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -298,13 +298,35 @@ please follow this procedure:
+* Install the missing Oracle components
+* execute the SQL scripts to *install* XA management elements
+* execute the SQL requests to GRANT the proper rights to the Oracle user
+
```
### Check the Bonita 7.8.4 server starts with the Oracle database server 12c (12.2.0.x.y)

* Download and install a Bonita 7.8.4 server
* Setup the Bonita 7.8.4 server to use the Oracle 12c (12.2.0.x.y) database
* Request and install a temporary 7.8 license in the Bonita server
* Start the Bonita 7.8.4 server
* Check you can successfully log into the portal

ugaston commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -298,13 +298,35 @@ please follow this procedure:
+
+* shut down the Bonita platform 
+* Run the migration tool to migrate the platform to 7.9+, following the procedure above
+* install target Tomcat application server with bonita 
```
... Is it necessarily a Tomcat, in 7.9+?
I believe we could still have Wildfly in 7.9.x (however deprecated it may have been)...

ugaston reviewed (commented):
Awesome!
Just a minor correction regarding the Oracle db version (I believe the supported versions are 12.2.0.x.y), a small addition to remind users to test the upgrade to Oracle 12.2.0.x.y went fine by having them check a 7.8.4 server can actually be started upon the upgraded database.

And a tiny general comment: your bullet points sometimes start with a lower case, and sometimes they do so with an upper case...

Cheers!

jjargot committed:
Update md/migrate-from-an-earlier-version-of-bonita-bpm.md

thx!

Co-Authored-By: ugaston <unai.gaston.caminos@bonitasoft.com>

jjargot committed:
Update md/migrate-from-an-earlier-version-of-bonita-bpm.md

thx!

Co-Authored-By: ugaston <unai.gaston.caminos@bonitasoft.com>

jjargot committed:
fix(migration): detail the Oracle version change

unify the case of the chars

jjargot committed:
fix(migration): detail the Oracle version change

unify the case of the chars

jjargot committed:
fix(migration): detail the Oracle version change

fix 6?

jjargot committed:
fix(migration): detail the Oracle version change

fix 07

Commit:
fix(migration): updates oracle

- Changed : in a first step, run Bonita migration tool to update Bonita platform to version 7.8.4, following the procedure [above](#migrate)
- Changed : run the migration tool to migrate the platform to 7.9+, following the procedure [above](#migrate)
- Removed: the following lines, it's already in the migration procedure : 
install target application server with bonita
etc.
- Changed (I'm not super satistied with that): 
step 4: If you use Oracle, and you are migrating from a version lower than 7.9.0, skip to the section Migrating to Bonita 7.9+ using Oracle below. If you use Oracle and are migrating from version 7.9.0 and higher, add the JDBC driver for your database to bonita-migration/lib. This is the same driver as you have installed in your web server lib directory. You must upgrade to Oracle 12c (12.2.x.y) in order to migrate to 7.9+.
- Fix: the migration full procedure numbering that was broken, based on https://stackoverflow.com/questions/18088955/markdown-continue-numbered-list

jjargot committed:
fix(migration): detail the Oracle version change

shuffle the perfect Marielle's new lines at point 4.

jjargot committed:
fix(migration): detail the Oracle version change

Repair the numbering and testing .. at the end of lines to have multi-line in the numbering

jjargot committed:
fix(migration): detail the Oracle version change

thx to Unai and Céline, I could use multi-line in a list with markdown

jjargot committed:
fix(migration): detail the Oracle version change

remove 4 blank chars and validate ok with Unai

pabloalonso reviewed (changes requested):
We plan to support Oracle19c in 7.9 and 7.10 
https://github.com/bonitasoft/bonita-doc/pull/1155
This PR should be updated accordling

jjargot committed:
fix(migration): detail the Oracle version change

With Unai, we injected the information for Oracle 19c (19.3.0.0)

Versions: 7.9, 7.10

jjargot commented:
Thank you Pablo, we've added the Oracle 19c (19.3.0.0) information.

marielle-spiteri commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -298,13 +301,50 @@ please follow this procedure:
+
+Each file however is specific to the Oracle DB server version installed.
+Please make sure to download the appropriate one:
+* Oracle 12c (12.2.0.x.y) : [Oracle Database 12.2.0.1 JDBC Driver & UCP Downloads](https://www.oracle.com/database/technologies/jdbc-ucp-122-downloads.html)
```
```suggestion
* Oracle 12c (12.2.0.x.y) : Driver ojdbc8.jar [Oracle Database 12.2.0.1 JDBC Driver & UCP Downloads](https://www.oracle.com/database/technologies/jdbc-ucp-122-downloads.html) ( make sure it is the official driver by checking the SHA1 Checksum: 60f439fd01536508df32658d0a416c49ac6f07fb )
```

marielle-spiteri commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -298,13 +301,50 @@ please follow this procedure:
+
+### Download the specific jdbc driver for the Oracle 12c (12.2.0.x.y) or 19c (19.3.0.0)
+
+**Beware**: two different jdbc driver jar files may share the same name (ojdbc8.jar for example).
```
```suggestion
**Beware**: two different jdbc driver jar files may share the same name (ojdbc8.jar).
```

marielle-spiteri commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -298,13 +301,50 @@ please follow this procedure:
+Each file however is specific to the Oracle DB server version installed.
+Please make sure to download the appropriate one:
+* Oracle 12c (12.2.0.x.y) : [Oracle Database 12.2.0.1 JDBC Driver & UCP Downloads](https://www.oracle.com/database/technologies/jdbc-ucp-122-downloads.html)
+* Oracle 19c (19.3.0.0) : [Oracle Database 19c (19.3) JDBC Driver & UCP Downloads](https://www.oracle.com/database/technologies/appdev/jdbc-ucp-19c-downloads.html)
```
```suggestion
* Oracle 19c (19.3.0.0) : Driver ojdbc8.jar [Oracle Database 19c (19.3) JDBC Driver & UCP Downloads](https://www.oracle.com/database/technologies/appdev/jdbc-ucp-19c-downloads.html) ( make sure it is the official driver by checking the SHA1 Checksum: 967c0b1a2d5b1435324de34a9b8018d294f8f47b )
```

educhastenier replied:
@jjargot please note that ojdbc8-19.3.0.0.jar is now provided in the migration tool.
Can you update this PR accordingly?

baptistemesta commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -146,11 +149,11 @@ Specify the following information:
+    Note: If you are using MySQL, add `?allowMultiQueries=true` to the URL. For example,
+    `db.url=jdbc:mysql://localhost:3306/bonita_migration?allowMultiQueries=true`.  
+    Also, if you are migrating to Bonita 7.9+, you must upgrade your database server to MySQL 8.0, see [Migrating to Bonita 7.9+ using MySQL](#mysql8) specific procedure below.
+    :::
```
```suggestion
::: info
Note: If you are using MySQL, add `?allowMultiQueries=true` to the URL. For example,
`db.url=jdbc:mysql://localhost:3306/bonita_migration?allowMultiQueries=true`.  
Also, if you are migrating to Bonita 7.9+, you must upgrade your database server to MySQL 8.0, see [Migrating to Bonita 7.9+ using MySQL](#mysql8) specific procedure below.
:::
```

marielle-spiteri reviewed (commented):
my review, please check

educhastenier commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -133,11 +133,14 @@ This section explains how to migrate a platform that uses one of the Bonita bund
+1. If you use Oracle, and you are migrating from a version lower than 7.9.0, you must upgrade to [Oracle 12c (12.2.0.x.y)](migrate-from-an-earlier-version-of-bonita-bpm.md#oracle12) in order to migrate to 7.9+.
+Please skip to the section [Migrating to Bonita 7.9+ using Oracle](#oracle12) below.
+
+    If you use Oracle and are migrating from version 7.9.0 and higher, add the JDBC driver for your database to `bonita-migration/lib`. This is the same driver as you have installed in your web server `lib` directory.  
```
Now, lib/ojdbc8-19.3.0.0.jar is provided with the migration tool.
Please update the documentation accordingly @jjargot . Thx.

educhastenier committed:
restore previous § formatting

Co-authored-by: Baptiste Mesta <baptiste.mesta@bonitasoft.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/bonitasoft/bonita-doc?pullRequest=1151) <br/>All committers have signed the CLA.

jjargot committed:
Update md/migrate-from-an-earlier-version-of-bonita-bpm.md

+1

Co-authored-by: Marielle Spiteri <marielle.spiteri@bonitasoft.com>

jjargot committed:
Update md/migrate-from-an-earlier-version-of-bonita-bpm.md

+1

Co-authored-by: Marielle Spiteri <marielle.spiteri@bonitasoft.com>

jjargot committed:
Update md/migrate-from-an-earlier-version-of-bonita-bpm.md

+1

Co-authored-by: Marielle Spiteri <marielle.spiteri@bonitasoft.com>

jjargot committed:
integration of Emmanuel remarks

done

jjargot committed:
Merge branch '7.9' into jjargot-migration-doc-1

jjargot committed:
Final general fixes

fixed

jjargot committed:
genral review last version

all changes merge, conflicts solved and remarks included

educhastenier commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -388,11 +392,52 @@ please follow this procedure:
+* Oracle 12c (12.2.0.x.y) : Driver ojdbc8.jar [Oracle Database 12.2.0.1 JDBC Driver & UCP Downloads](https://www.oracle.com/database/technologies/jdbc-ucp-122-downloads.html) ( make sure it is the official driver by checking the SHA1 Checksum: 60f439fd01536508df32658d0a416c49ac6f07fb )
+* Oracle 19c (19.3.0.0) : Driver ojdbc8.jar [Oracle Database 19c (19.3) JDBC Driver & UCP Downloads](https://www.oracle.com/database/technologies/appdev/jdbc-ucp-19c-downloads.html) ( make sure it is the official driver by checking the SHA1 Checksum: 967c0b1a2d5b1435324de34a9b8018d294f8f47b )
+
+**Note I**: The migration tool already include the oracle driver for Oracle 19c (19.3.0.0) in the `bonita-migration/lib` directory. If your are not using Oracle 19c (19.3.0.0) you need to replace it.
```
```suggestion
**Note I**: The migration tool already includes the oracle driver for Oracle 19c (19.3.0.0) in the `bonita-migration/lib` directory. If your are not using Oracle 19c (19.3.0.0) you need to replace it.
```

educhastenier committed:
minor syntax fix
